stactask/payload.py

```diff
@@ -0,0 +1,168 @@
+                    )
+            return task_options
+
+    @property
+    def items_as_dicts(self) -> list[dict[str, Any]]:
+        features = self.get("features", [])
+        if isinstance(features, list):
+            return features
+        else:
+            raise TypeError("unable to parse 'features': must be type list")
+
+    @property
+    def global_upload_options(self) -> dict[str, Any]:
+        upload_options = self.process_definition.get("upload_options", {})
+        if isinstance(upload_options, dict):
+            return upload_options
+        else:
+            raise TypeError("unable to parse 'upload_options': must be type dict")
+
+    @property
+    def collection_mapping(self) -> dict[str, str]:
+        collection_mapping = self.global_upload_options.get("collections", {})
+        if isinstance(collection_mapping, dict):
+            warnings.warn(
+                (
+                    "'upload_options.collections' is deprecated and will be removed in "
+                    "a future version. Use 'collection_matchers' instead."
+                ),
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            return collection_mapping
+        else:
+            raise TypeError("unable to parse 'collections': must be type dict")
+
+    @property
+    def collection_matchers(self) -> list[dict[str, Any]]:
+        matchers = self.process_definition.get("collection_matchers", [])
+        if not isinstance(matchers, list):
+            raise TypeError("unable to parse 'collection_matchers': must be type list")
+        if not all(isinstance(matcher, dict) for matcher in matchers):
+            raise TypeError(
+                "unable to parse 'collection_matchers': each matcher must be type dict"
+            )
+        return matchers
+
+    @property
+    def collection_options(self) -> dict[str, dict[str, Any]]:
+        options = self.process_definition.get("collection_options", {})
+        if not isinstance(options, dict):
+            raise TypeError("unable to parse 'collection_options': must be type dict")
+        return options
+
+    def get_collection_options(self, collection_name: str) -> dict[str, Any]:
+        options = self.collection_options.get(collection_name, {})
+        if not isinstance(options, dict):
+            raise TypeError(
+                f"unable to parse 'collection_options' for collection "
+                f"'{collection_name}': must be type dict"
+            )
+        return options
+
+    def get_collection_upload_options(self, collection_name: str) -> dict[str, Any]:
+        options = (
+            self.get_collection_options(collection_name).get("upload_options", {})
+            or self.global_upload_options
+        )
+        if not options:
+            raise ValueError(
+                f"No upload options found for collection '{collection_name}'"
+            )
```

Comment:
It's valid not to have `upload_options`, yeah? I think this check should be removed.

Reply:
Yes, that is correct insofar as that it will not cause any downstream errors. I added that `raise` based on:

>if any collections do not have upload_options then check for global upload options, and if that is missing throw an error

from #124. 

But I think it is OK for a collection to not have upload options. I'll remove it for now but leave this comment unresolved in case that issue description is indicative of something pertinant.

Reply:
I had a misunderstanding of the current behavior when I wrote that. Sorry to mislead you.